### ISSUE TITLE
compress bundles for long-term storage

### ISSUE THREAD

**Issue by percyliang** [codalab/codalab-worksheets#297]
Compress bundle directories to save space (use tar.gz).

Unfortunately, tar.gz doesn't allow for random access.  Should keep an index into the tar file so we can see quickly what's in the tar.  Can we get random access?  One option is to go to zip, but the rest of the system uses tar, so it'd be better to stick with that.

Consideration: directives need to pull out parts of the bundle (not necessarily known in advance).  Heuristic: leave the first 1K or so each file uncompressed?  This might be useful even if we have random access because we'll need it if files are stored in S3.

Policy for compressing: some daemon in the back should be doing this to bundles that don't seem like they are used that much, or just after a certain period of time.  Or we could even just compress everything.  (When files are transferred, they are compressed anyway.)

In summary, we provide only two ways to store a bundle:
1) tar.gz (.gz for uploaded files)
2) summary: first N and last N lines of each file (max # files = 1000), and a full file listing
3) untarred version - this should only be on the worker machine

Worker should be updating 2) all along, and then should transfer 1) when everything is said and done.  When transferring 2), should transfer diff?

**Comment by fabeschan** [comment]
Compression seems to only save 40% of space. Depending on the storage backend we choose the amount of complexity might not be worth it. We need to tie this in with whatever design we come up with for bundle storage 2.0 before we make a decision

**Comment by percyliang** [comment]
So it doesn't seem worth it, closing for now.  One related problem is that if a bundle has tons of files and the bundle store is on a network file system, it's really slow...
